qt-gui: Added GPU device selection functionality

SamuelFontes opened the pull request:
This is a good solution for the problem people are having on the issue: https://github.com/shadps4-emu/shadPS4/issues/232
![image](https://github.com/user-attachments/assets/2c19666a-9d5a-4eed-b8fe-912317370fd3)

SamuelFontes committed:
qt-gui: Added GPU device selection functionality

SamuelFontes committed:
Getting list of GPU only when application starts

SamuelFontes committed:
Fixed formatting

SamuelFontes committed:
Fixed formatting

SamuelFontes committed:
Fixed formatting

SamuelFontes committed:
Added warning when GPU doesn't support API version.

SamuelFontes committed:
Changed Unsupported Vulkan Version warning

raphaelthegreat commented on src/qt_gui/main_window.cpp:
```diff
@@ -158,6 +160,21 @@ void MainWindow::LoadGameLists() {
 }
 
+void MainWindow::GetPhysicalDevices() {
+    if (m_physical_devices.size() == 0) {
```
If main window init is called once the size check here is probably unnecessary

SamuelFontes replied:
Forgot to remove that, thanks for pointing it out.
It was some left over of me trying to get this done some other way.

raphaelthegreat commented on src/qt_gui/settings_dialog.cpp:
```diff
@@ -4,13 +4,20 @@
 
+    // Add list of available GPUs
+    ui->graphicsAdapterBox->addItem("Auto Select"); // -1, auto selection
+    for (auto device = physical_devices->begin(); device != physical_devices->end(); ++device) {
```
```suggestion
    for (const auto& device : physical_devices) {
```

raphaelthegreat commented on src/qt_gui/settings_dialog.h:
```diff
@@ -16,7 +16,8 @@ class SettingsDialog;
 public:
-    explicit SettingsDialog(QWidget* parent = nullptr);
+    explicit SettingsDialog(QWidget* parent = nullptr,
+                            std::vector<QString>* physical_devices = nullptr);
```
```suggestion
                            std::span<const QString> physical_devices);
```

SamuelFontes replied:
I've tried doing that way but for some reason it doesn't compile due to an error: missing default argument on parameter 'physical_devices'
![image](https://github.com/user-attachments/assets/1bd1fd9d-f50e-4b58-9465-409e0220783b)
Not sure if this is something on my environment only

raphaelthegreat replied:
That is because the first function argument has a default argument. Default argument are only supported at end of list. Either remove default or move std::span as first arg

SamuelFontes replied:
Thanks that solved the issue.

SamuelFontes committed:
Removed unused size checking on GetPhysicalDevices

The method is only being called once so this doesn't make sense. It was some left over of me trying to get this done some other way.

SamuelFontes committed:
Fix formatting

SamuelFontes committed:
Fix formatting

SamuelFontes committed:
SettingsDialog: Passing physical devices as span

SamuelFontes committed:
Fixed formatting

raphaelthegreat reviewed (approved):
Thanks
